test/helper/sending.bash

```diff
@@ -45,57 +47,127 @@ function _send_email() {
+      ( * ) ADDITIONAL_SWAKS_OPTIONS+=("'${1}'") ; shift 1 ;;
     esac
   done
 
-  _run_in_container_bash "swaks --server ${SERVER} --port ${PORT} --ehlo ${EHLO} --from ${FROM} --to ${TO} ${ADDITIONAL_SWAKS_OPTIONS[*]} ${FINAL_SWAKS_OPTIONS[*]}"
+  _run_in_container_bash "swaks --server '${SERVER}' --port '${PORT}' --ehlo '${EHLO}' --from '${FROM}' --to '${TO}' ${ADDITIONAL_SWAKS_OPTIONS[*]}"
+}
+
+# Sends a mail from localhost (127.0.0.1) to a container. To send
+# a custom email, create a file at `test/files/<TEST FILE>`,
+# and provide `<TEST FILE>` as an argument to this function.
+#
+# Parameters include all options that one can supply to `swaks`
+# itself. The `--data` parameter expects a relative path from `emails/`
+# where the contents will be implicitly provided to `swaks` via STDIN.
+#
+# This functions performs an implicit `assert_success` to check whether
+# the e-mail transaction was successful. If this is not desirable, use
+# `_send_email_unchecked`.
+#
+# ## Attention
+#
+# This function assumes `CONTAINER_NAME` to be properly set (to the container
+# name the command should be executed in)!
+#
+# This function will just send the email in an "asynchronous" fashion, i.e. it will
+# send the email but it will not make sure the mail queue is empty after the mail
+# has been sent.
+function _send_email() {
+  _send_email_unchecked "${@}"
+  assert_success
 }
 
 # Like `_send_email` with two major differences:
 #
 # 1. this function waits for the mail to be processed; there is no asynchronicity
-#    because filtering the logs in a synchronous way is easier and safer!
-# 2. this function prints an ID one can later filter by to check logs
+#    because filtering the logs in a synchronous way is easier and safer;
+# 2. this function takes the name of a variable and inserts ID(s) one can later
+#    filter by to check logs.
 #
 # No. 2 is especially useful in case you send more than one email in a single
 # test file and need to assert certain log entries for each mail individually.
 #
-# This function takes the same arguments as `_send_mail`.
+# The first argument has to be the name of the variable that the e-mail ID is stored
+# in. The second argument **can** be the flag `--unchecked`; if this flag is supplied,
+# the function uses `_send_email_unchecked` instead of `_send_email`. This avoids the
+# `assert_success`. Be warned though this is only required in special situations where
+# it is still possible to `grep` for the mail IDs - otherwise this function fails.
+#
+# The rest of the arguments are the same as `_send_email`.
 #
 # ## Attention
 #
 # This function assumes `CONTAINER_NAME` to be properly set (to the container
 # name the command should be executed in)!
 #
+# Moreover, if `--data <DATA>` is specified, the additional header added implicitly
+# (with `--add-header`) may get lost, so pay attention to the data having the token
+# to place additonal headers into.
```

Comment:
Verify if still applicable with `--header`?

Reply:
Yes, this still holds, because I wrote the fix above so you can still supply "inline" data.